src/main/java/core/basesyntax/service/impl/ShopServiceImpl.java

```diff
@@ -0,0 +1,32 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.operations.OperationHandler;
+import core.basesyntax.operations.OperationStrategy;
+import core.basesyntax.service.ShopService;
+import java.util.List;
+import java.util.Map;
+
+public class ShopServiceImpl implements ShopService {
+    private final OperationStrategy operationStrategy;
+    private final Storage storage;

```

Comment:
According to the encapsulation principle, direct access to the `Storage` should be avoided. Instead of passing the `Storage` directly, consider using a `StorageService` that encapsulates the logic for interacting with the storage.